src/jit/lower.cpp

```diff
@@ -1330,9 +1328,9 @@ void Lowering::LowerArg(GenTreeCall* call, GenTree** ppArg)
             GenTree* argLo = arg->gtGetOp1();
             GenTree* argHi = arg->gtGetOp2();
```

Comment:
Need to take a closer look at all this code, I'm not quite sure what it is doing. In both case a field list and a put arg are created but in the reg case the list is not inserted in LIR and `*ppArg` ends up pointing directly to the list instead of the newly created put arg node. I guess that's actually dead code because `long` on x86 is treated as a struct so it always ends up on the stack, not in registers.

Reply:
Ah, `NewPutArg` is actually returning the original field list in the reg case. Well, that's confusing :)